### ISSUE TITLE
Four confused questions.

### ISSUE THREAD

**Issue by shoutOutYangJie** [saic-vul/adaptis#11]
1.  

>   if not math.isclose(self.spatial_scale, 1.0):
            instance_out = F.interpolate(instance_out, orig_size, mode='bilinear', align_corners=True)

self.spatial_scale is 1, so it doesn't run bilinear upsampling. so final output is 4 times smaller that original image.  And when calculating loss, there is no upsampling layer, so how could using final output and GT to calculate loss?

2. where is "bilinear feature extractor" in your code?
![image](https://user-images.githubusercontent.com/46734514/68280352-53e54580-00b0-11ea-893c-7b59f3804781.png)

3.  given points belong to  the  coordinate of original image, and in your code, you didn't divide it by 4, so is it still used to ROIalign?

4. similarl with last quesition, the coordinate of points is the same as original image, why can it be sent to Relative coordconv Block? 

>  relative_coords = (coords - add_xy) / (self.norm_radius * self.spatial_scale)

here coords is at the coordinate of the feature from backbone, but add_xy comes from original image.

**Comment by shoutOutYangJie** [comment]
the sixth question.
6. 

>   if self.with_proposals:
            backbone_features = backbone_features.detach()
            proposals_out = self.proposals_head(backbone_features)
            proposals_out = self.adaptis_head.EQF(proposals_out, points.detach())
            outputs.append(('proposals', proposals_out))

the proposal branch outputs a vector extracted by ROI align? really? and then this vector is sent to loss.
But, when calculating loss, you use instance mask as GT?

>  loss = self._add_loss('proposals_loss', loss, losses_logging, validation,
                                  lambda: (output.instances, output.proposals, batch_data['instances'].to(self.device)))

Jesus, I am confused.

**Comment by ksofiyuk** [comment]
1. The output of the instance segmentation branch is upsampled to image scale inside forward method. Look at [line 50](https://github.com/saic-vul/adaptis/blob/4a973e81e936494a98f10b7f6dbf1f4fd4ad1328/adaptis/model/adaptis.py#L50)

**Comment by shoutOutYangJie** [comment]
Thanks for your rapid reply.
1. what I mean is , at your code. you set spatial_scale as 1. So, **math.isclose(self.spatial_scale ,1)** is True,  and then you add **'not'** before  **math.isclose(self.spatial_scale ,1)**, so code will not run **'instance_out = F.interpolate(instance_out, orig_size, mode='bilinear', align_corners=True)'**

7, I add a new question. What is **rescale_output=(0.2, -1.7)** used to do .
>  if self.rescale_output:
            scale, bias = self.rescale_output
            x = scale * x + bias

why do you do this?

**Comment by ksofiyuk** [comment]
1. All right, when the spatial scale is 1 output size of the backbone should match the image size. In this case, we don't need to interpolate.
7. It improves initialization for Normalized Focal Loss and it is used only for the toy model. With these changes, the model begins to converge more smoothly at the beginning.
